common-service-core/src/main/java/com/ca/mfaas/message/core/Message.java

```diff
@@ -0,0 +1,167 @@
+
+import java.util.Collections;
+import java.util.Objects;
+import java.util.UUID;
+
+/**
+ * Message creator immutable class
+ */
+public final class Message {
+
+    public static final String INVALID_KEY_MESSAGE = "com.ca.mfaas.common.invalidMessageKey";
+    public static final String INVALID_MESSAGE_TEXT_FORMAT = "com.ca.mfaas.common.invalidMessageTextFormat";
+
+    private final String requestedKey;
+    private final MessageTemplate messageTemplate;
+    private final Object[] messageParameters;
+
+
+    private Message(String requestedKey,
+                    MessageTemplate messageTemplate,
+                    Object[] messageParameters) {
+        this.requestedKey = requestedKey;
+        this.messageTemplate = messageTemplate;
+        this.messageParameters = messageParameters;
+    }
+
+    /**
+     * Return a {@link Message} object for the specified key after text and parameters validation.
+     *
+     * @param requestedKey the message key.
+     * @param messageTemplate the messageTemplate.
+     * @param messageParameters the object containing the message parameters.
+     * @return {@link Message}
+     */
+    public static Message of(String requestedKey,
+                             MessageTemplate messageTemplate,
+                             Object[] messageParameters) {
+
+        requireNotNull(requestedKey, "requestedKey can't be null");
+        requireNotNull(messageTemplate, "messageTemplate can't be null");
+        requireNotNull(messageParameters, "messageParameters can't be null");
+
+        validateMessageTextFormat(messageTemplate.getText(), messageParameters);
+        messageParameters = validateParameters(messageTemplate.getKey(), requestedKey, messageParameters);
+
+        return new Message(requestedKey, messageTemplate, messageParameters);
+    }
+
+
+    /**
+     * Checks that the specified object reference is not null and
+     * throws a {@link IllegalArgumentException} if it is.
+     *
+     * @param param the object reference to check for nullity
+     * @param message detail message to be used in the event
+     */
+    private static void requireNotNull(Object param, String message) {
+        if (param == null) {
+            throw new  IllegalArgumentException(message);
+        }
+    }
+
+    /**
+     * Validate the message text and parameters returning them as formatted String.
+     *
+     * @param messageText the message text.
+     * @param messageParameters the object containing the message parameters.
+     * @return a formatted String
+     */
+    private static String validateMessageTextFormat(String messageText, Object[] messageParameters) {
+        return String.format(messageText, messageParameters);
+    }
+
+    /**
+     * Check if the key is equal to the invalid key message. If it is, the parameter is set to the requested key,
+     * which will be displayed in the invalid key message text, otherwise the passed parameters are returned.
```

Comment:
Ok, this one is tricky to decipher. Let me ask some questions: 
Once the key doesn't match (or no equal to) the invalid key message, what exactly will happen? According to the aforementioned, "the parameter" will be set to "the requested key". Now, what does that mean? What will be displayed in the invalid ket message text? "The parameter" or "the requested key"?